### PR TITLE
Request Missing HighQC Block For Timeout Message

### PR DESCRIPTION
**Pull request by tholonious**

### CODE REVIEW

**Comment by tholonious** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1061?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#1061** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1061?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1058** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1058?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1057** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1057?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/pos-networking-and-syncing`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1061?utm_source=stack-comment).

File: lib/pos_consensus.go [hunk 1]
```diff
@@ -507,8 +507,28 @@ func (fc *FastHotStuffConsensus) HandleLocalTimeoutEvent(event *consensus.FastHo
...
+
+	if !fc.fastHotStuffEventLoop.IsRunning() {
+		return errors.Errorf("FastHotStuffConsensus.HandleValidatorTimeout: FastHotStuffEventLoop is not running")
+	}
+
+	// Hold the blockchain's read lock so that the chain cannot be mutated underneath us.
+	// In practice, this is a no-op, but it guarantees thread-safety in the event that other
+	// parts of the codebase change.
+	fc.blockchain.ChainLock.Lock()
+	defer fc.blockchain.ChainLock.Unlock()
```
Comment (lazynina) [inline lib/pos_consensus.go:522]: Should we release `fc.blockchain.Chainlock` and `fc.lock` before we call `fc.fastHotStuffEventLoop.ProcessValidatorTimeout`? My thinking is that we only need to hold the lock while we are checking the highQC and that `fc.fastHotStuffEventLoop.ProcessValidatorTimeout` should handle locks on its own?

Comment (tholonious) [reply lib/pos_consensus.go:522]: Sure. We're find to re-jig the ordering of the locks so they're held for the minimum amount of time

Comment (tholonious) [reply lib/pos_consensus.go:522]: We need to guarantee that the handlers in this file are single threaded, which is why we hold fc.lock at the beginning. We could tune things to be more efficient by reordering the locks for less time, but this comes at the cost of deadlock risk. IMO it's not worth optimizing right now.

In this case, it's a great idea to hold and release the fc.blockchain only while performing the blockIndex lookup. I'll update to that 👍

**Review by lazynina — APPROVED** [review]

**Pull request merged**